src/components/legend/style.js

```diff
@@ -0,0 +1,216 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var d3 = require('d3');
+
+var Lib = require('../../lib');
+var Plots = require('../../plots/plots');
+var Drawing = require('../drawing');
+var Color = require('../color');
+
+var subTypes = require('../../traces/scatter/subtypes');
+var stylePie = require('../../traces/pie/style_one');
+
+
+module.exports = function style(s) {
+    s.each(function(d) {
+        var traceGroup = d3.select(this);
+
+        var fill = traceGroup
+            .selectAll('g.legendfill')
+                .data([d]);
+        fill.enter().append('g')
+            .classed('legendfill',true);
+
+        var line = traceGroup
+            .selectAll('g.legendlines')
+                .data([d]);
+        line.enter().append('g')
+            .classed('legendlines',true);
+
+        var symbol = traceGroup
+            .selectAll('g.legendsymbols')
+                .data([d]);
+        symbol.enter().append('g')
+            .classed('legendsymbols',true);
+        symbol.style('opacity', d[0].trace.opacity);
+
+        symbol.selectAll('g.legendpoints')
+            .data([d])
+          .enter().append('g')
+            .classed('legendpoints',true);
+    })
+    .each(styleBars)
+    .each(styleBoxes)
+    .each(stylePies)
+    .each(styleLines)
+    .each(stylePoints);
+};
+
+function styleLines(d) {
+    var trace = d[0].trace,
```

Comment:
these should really be methods of the corresponding trace module, but that will be for a future PR.